source/extensions/filters/http/ext_proc/ext_proc.cc

```diff
@@ -140,11 +140,35 @@ FilterConfig::initExpressions(const Protobuf::RepeatedPtrField<std::string>& mat
 
 FilterConfigPerRoute::FilterConfigPerRoute(const ExtProcPerRoute& config)
     : disabled_(config.disabled()) {
-  if (config.has_overrides()) {
+  if (!config.has_overrides()) {
+    return;
+  }
+
+  const auto& overrides = config.overrides();
+  if (overrides.has_processing_mode()) {
     processing_mode_ = config.overrides().processing_mode();
   }
-  if (config.overrides().has_grpc_service()) {
-    grpc_service_ = config.overrides().grpc_service();
+  if (overrides.has_grpc_service()) {
+    grpc_service_ = overrides.grpc_service();
+  }
+
+  if (!overrides.has_metadata_options()) {
```

Comment:
I would suggest moving all metadata initialization code below #159-#172 into a separate function and just make a call instead, as this exit condition will have to be replaced as soon as any additional logic will be required down there to be added.

Reply:
Thanks for the suggestion @sitano 

This will be refactored into the ctor list once https://github.com/envoyproxy/envoy/pull/31431 merges